helper/locale.js

```diff
@@ -0,0 +1,94 @@
+module.exports = {
+
+  // returns the correct supported locale given the LANG variable
+  getLocale() {
+    const angularConf = require('../angular.json');
+    const i18n = angularConf.projects.OctoDash.i18n;
+    let lang;
+    try {
+      // Detect if environment LANG is set
+      // if it is defined, convert LANG from ie 'en_US.UTF-8' to 'en-US'
+      // if not defined, set it to the source locale
+      lang = process.env['LANG']
+        ? process.env['LANG'].split('.')[0].replace('_', '-')
+        : i18n.sourceLocale.code;
+    } catch(e) {
+      // LANG was populated with something else than a standard locale code
+      lang = i18n.sourceLocale.code;
+    }
+    // Check if exact LANG exists in our locales
+    const exactLocale = Object.keys(i18n.locales).includes(lang) && lang;
+    // Make the short version, convert ie 'en-US' to 'en'
+    const shortLang = lang.split('-')[0];
+    // Check if short LANG exists in our locales
+    // This matches 'fr-CA' to 'fr' if 'fr-CA' is not defined but 'fr' is
+    const approximateLocale = Object.keys(i18n.locales).includes(shortLang) && shortLang;
+    // Define locale by either the exact, approximate, or source locale in this order
+    const locale = exactLocale || approximateLocale || i18n.sourceLocale.code;
```

Comment:
this file is javascript, not transpiled by angular's webpack

Reply:
It's supported by JS as well: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
TIL `??` and `.?` are now a thing in JS! That's amazing. I need to double-check the logic with ??, because the simple ORs work with empty string and i think it was a possible case. I mean, ?? may be fancier, but the || in this case makes sense, we don't want an empty string, false bool or anything to be a valid value.

Reply:
Ah ok if this can be an empty string the OR operator will be better. Also says that in the top of docs that `??` only matches `null` or `undefined`